src/cljs-lib/src/calva/repl/webview/core.cljs

```diff
@@ -0,0 +1,228 @@
+(ns calva.repl.webview.core
+  (:require
+   [calva.util :as util]
+   [clojure.string :as str]))
+
+(defonce repl-output-webview-panel (atom nil))
+
+(defn dispose-repl-output-webview-panel
+  [webview-panel-atom]
+  (reset! webview-panel-atom nil))
+
+(defn post-message-to-webview [^js webview-panel message]
+  (when webview-panel
+    (.. webview-panel
+        -webview
+        (postMessage (pr-str (merge
+                              {:id (str (random-uuid))} ;; Provide an id if one wasn't provided by the caller
+                              message))))))
+
+;; The connect-src and unsafe-eval are only needed in development mode for the shadow-cljs
+;; dev workflow to function properly
+(defn get-webview-html
+  [{:env/keys [is-debug]} {:keys [js-source css-href csp-source]}]
```

Comment:
A thing I have started to do is to “namespace” all keys I get away with doing it. So here, for instance, I would invent a namespace, say `:html/` for the args. It gets you rename support and often the same maps are bing sent around and you can find references and such.

Just noting. Nothing we need to address in any hurry, me thinks. (I am not even sure it is a good idea, even if I have had pretty good experience from it.)